Lib/test/regrtest.py

```diff
@@ -509,58 +547,34 @@ def main(tests=None, **kwargs):
 
     ns = _parse_args(sys.argv[1:], **kwargs)
 
-    if ns.huntrleaks:
-        # Avoid false positives due to various caches
-        # filling slowly with random data:
-        warm_caches()
-    if ns.memlimit is not None:
-        support.set_memlimit(ns.memlimit)
-    if ns.threshold is not None:
-        import gc
-        gc.set_threshold(ns.threshold)
-    if ns.nowindows:
-        print('The --nowindows (-n) option is deprecated. '
-              'Use -vv to display assertions in stderr.')
-    try:
-        import msvcrt
-    except ImportError:
-        pass
-    else:
-        msvcrt.SetErrorMode(msvcrt.SEM_FAILCRITICALERRORS|
-                            msvcrt.SEM_NOALIGNMENTFAULTEXCEPT|
-                            msvcrt.SEM_NOGPFAULTERRORBOX|
-                            msvcrt.SEM_NOOPENFILEERRORBOX)
-        try:
-            msvcrt.CrtSetReportMode
-        except AttributeError:
-            # release build
-            pass
-        else:
-            for m in [msvcrt.CRT_WARN, msvcrt.CRT_ERROR, msvcrt.CRT_ASSERT]:
-                if ns.verbose and ns.verbose >= 2:
-                    msvcrt.CrtSetReportMode(m, msvcrt.CRTDBG_MODE_FILE)
-                    msvcrt.CrtSetReportFile(m, msvcrt.CRTDBG_FILE_STDERR)
-                else:
-                    msvcrt.CrtSetReportMode(m, 0)
-    if ns.wait:
-        input("Press any key to continue...")
-
     if ns.slaveargs is not None:
-        args, kwargs = json.loads(ns.slaveargs)
-        if kwargs.get('huntrleaks'):
+        ns_dict, testname = json.loads(ns.slaveargs)
+        ns = types.SimpleNamespace(**ns_dict)
+
+        setup_tests(ns)
+
+        if ns.huntrleaks:
             unittest.BaseTestSuite._cleanup = False
+
         try:
-            result = runtest(*args, **kwargs)
+            result = runtest(testname, ns.verbose, ns.quiet,
```

Comment:
Oh, nice catch, fixed.

It's a bug in my change. I copied the runtest() call from above, but above, there is a first "support.use_resources = ns.use_resources" call.